src/include/sof/math/exp_fcn_hifi.h

```diff
@@ -0,0 +1,36 @@
+/* SPDX-License-Identifier: BSD-3-Clause
+ *
+ * Copyright(c) 2023 Intel Corporation. All rights reserved.
+ *
+ * Author: Shriram Shastry <malladi.sastry@linux.intel.com>
+ *
+ */
+#ifndef __SOFM_EXP_HIFI_H__
+#define __SOFM_EXP_HIFI_H__
+
+#include <sof/platform.h>
+#include <stdint.h>
+#include <stddef.h>
+
+#if defined(__XCC__)
+/* HiFi */
+#include <xtensa/config/core-isa.h>
+#if XCHAL_HAVE_HIFI5 == 1
+#define EXPONENTIAL_HIFI5 1
+
+#elif XCHAL_HAVE_HIFI4 == 1
+#define EXPONENTIAL_HIFI4 1
+
+#elif XCHAL_HAVE_HIFI3 == 1
+#define EXPONENTIAL_HIFI3 1
+#else
+#error "No XCHAL_HAVE_HIFIx found. Cannot build XCC EXP - HiFi module."
+#endif
+#else
+/* GCC */
```

Comment:
```suggestion
/* not XCC */
```
This could be clang or anything.